admin/spam/forms.py

```diff
@@ -0,0 +1,15 @@
+from django import forms
+from ckeditor.widgets import CKEditorWidget
+
+
+class EmailForm(forms.Form):
+    author = forms.CharField(label='Author', max_length=100)
+    email = forms.ChoiceField(label='Email')
+    subject = forms.CharField(label='Subject', required=True)
+    message = forms.CharField(label='Message', required=True,
+                              widget=CKEditorWidget())
+
+    def __init__(self, *args, **kwargs):
+        choices = kwargs.get('initial', {}).get('email', [])
+        self.base_fields['email'] = forms.ChoiceField(choices=choices)
```

Comment:
You can set the choices kwarg to a callable so you could do:
```python
class EmailForm(forms.Form):
    def _get_email_choices(self):
        return self.initial.get('email', [])
    email = forms.ChoiceField(label='Email', choices=self._get_email_choices)
```
https://github.com/django/django/blob/master/django/forms/forms.py#L81 says \__init__ sets self.initial to kwargs[initial].